Fix issue #19407:Blog page is not responsive on mobile devices

2008sahil opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

Responsive Fixes: This PR addresses issue #19407 by implementing fixes to enhance the responsiveness of the blog page. The changes ensure that the blog page displays correctly on various mobile devices, improving the overall user experience.

UI Enhancements: In addition to resolving the responsiveness issue, this PR incorporates suggested UI enhancements provided by Oppia community members. The improvements aim to make the blog page more visually appealing and user-friendly, aligning with the design standards of the Oppia project.

Optimized Layout: The layout adjustments introduced in this PR optimize the arrangement of elements on the blog page, ensuring a consistent and polished appearance across different screen sizes. These changes contribute to a more seamless navigation experience for users interacting with the blog on both desktop and mobile platforms.

## Essential Checklist

- [ ✓] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [✓ ] I have followed the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- [✓ ] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{seanlip}} PTAL" if I don't have permissions to assign reviewers directly).
- [ ✓] The linter/Karma presubmit checks pass on my local machine, and my PR follows the [coding style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide)).
- [✓] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)

## Testing doc (for PRs with Beam jobs that modify production server data)



- [✓ ] _(To be confirmed by the server admin)_ All jobs in this PR have been verified on a live server, and the PR is safe to merge.


## Proof that changes are correct

https://github.com/oppia/oppia/assets/155609546/1c0e6254-ad1b-4e7f-967d-92e80c2af29e


<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly.



When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

Commit:
fixed UI changes in blog page

Commit:
fixed UI changes in blog page

Commit:
passing lint testcase1

Commit:
passing lint testcase1

Commit:
passing lint testcase2

Commit:
passing lint testcase2

Commit:
all lint case passed

Commit:
all lint case passed

Commit:
all lint case passed

Commit:
all lint case passed

Commit:
final commit

Commit:
final commit

2008sahil committed:
Merge branch 'develop' into blogfix

seanlip commented on core/templates/pages/blog-home-page/blog-home-page.component.html:
```diff
@@ -86,7 +86,7 @@
           <source type="image/webp" [srcset]="getStaticImageUrl('/images/general/no_explorations_found.webp')">
           <source type="image/png" [srcset]="getStaticImageUrl('/images/general/no_explorations_found.png')">
-          <img [src]="getStaticImageUrl('/general/no_explorations_found.png')">
+          <img  [src]="getStaticImageUrl('/general/no_explorations_found.png')">
```
Drop the extra space before [src].

In general please look at the Files Changed tab before asking for a review, and ensure that the changes there correctly reflect the ones you want to make. Thanks.

2008sahil replied:
Sorry for that , I will  always keep it in mind.

seanlip commented on core/templates/pages/blog-home-page/blog-home-page.component.html:
```diff
@@ -114,20 +114,19 @@ <h4 class="d-flex flex-grow-1 search-heading">
           <form class="navbar-form oppia-search-bar-form pl-0 e2e-test-search-field" role="search">
             <div class="form-group d-flex"
-                 [ngClass]="{'flex-row-reverse': isSmallScreenViewActive()}">
+                 >
```
Move this to the end of the previous line.

2008sahil replied:
Done.

seanlip commented on core/templates/pages/blog-home-page/blog-home-page.component.html:
```diff
@@ -86,7 +86,7 @@
+          <img  [src]="getStaticImageUrl('/general/no_explorations_found.png')">
         </picture>
         <p class="text-center">
           {{ 'I18N_BLOG_HOME_PAGE_NO_RESULTS_FOUND' | translate }}
```
In the UI, please drop the per-word capitalization on this, it is not necessary.

2008sahil replied:
i have already removed the per-word capitalization property in the line no. 43 of core/templates/pages/blog-home-page/blog-home-page.component.css

2008sahil replied:
i have already removed the per-word capitalization property in the line no. 43 of core/templates/pages/blog-home-page/blog-home-page.component.css

2008sahil replied:
Done

seanlip commented on core/templates/pages/blog-home-page/blog-home-page.component.html:
```diff
@@ -86,7 +86,7 @@
         <picture>
```
Please add a bit more spacing using CSS as mentioned here: https://github.com/oppia/oppia/issues/19407#issuecomment-1890745921

2008sahil replied:
i have added spacing to the icon as well as no_result_found message in the line no. 46 should i make it 30 px, is it will be good ??.

2008sahil replied:
Done

2008sahil reviewed (commented):
I have done the suggested changes and updated the final result vedio .

Commit:
done suggested changes

Commit:
done suggested changes

Commit:
Merge remote-tracking branch 'upstream/develop' into blogfix

Commit:
Merge branch 'blogfix' of https://github.com/2008sahil/oppia into blogfix

kevintab95 commented on core/templates/pages/blog-home-page/blog-home-page.component.css:
```diff
@@ -116,17 +119,20 @@ oppia-blog-home-page .search-heading {
   color: #333;
 }
 
+
```
Please drop the newline.

2008sahil replied:
Done

kevintab95 reviewed (changes requested):
Just one nit. Thanks!

Commit:
Suggested Changes Done

Commit:
Suggested Changes Done

2008sahil commented:
@seanlip PTAL

seanlip reviewed (changes requested):
@2008sahil Thanks. I have one concern -- on the narrowest widths in your video, the grey border suddenly comes to the left and right of the box, giving it not enough space. Can we not have that?

Commit:
added media query at 470px

2008sahil commented:
@seanlip I have  added media query at 470px to eliminate current behaviour, here is the final result

https://github.com/oppia/oppia/assets/155609546/a5b7791e-8e6a-4f98-824f-9e7873bdc6f0

PTAL

seanlip commented:
@2008sahil In your last video, there's still a jump at the end, and the padding around the white box becomes significantly less. We should avoid that.

2008sahil committed:
Merge branch 'develop' into blogfix

2008sahil commented:
@seanlip , Since no blogs are displaying in running oppia  locally, I cann't find the effects of changing the width  of oppia-blog-home-page-card on blog posts So i tried inspecting on live websites here are the changes i observed we need to do :-
In oppia\core\templates\pages\blog-home-page\blog-home-page.component.css
1: Changing the width of class oppia-blog-home-page-card to 80%.
Before Changes:
![image](https://github.com/oppia/oppia/assets/155609546/347ff4bb-b8fe-4577-9c1c-0e7820fbd10b)
After Changes:
![image](https://github.com/oppia/oppia/assets/155609546/2b01a56f-4a31-4826-a6d1-e7a009cde298)

2:Changing the width of class blog-home-page-search-column to 100%.
Before Changes:
![image](https://github.com/oppia/oppia/assets/155609546/1d8734e1-55f2-4f8e-a6fa-0198a57fa71e)

After Changes:
![image](https://github.com/oppia/oppia/assets/155609546/78e49d34-9451-44aa-a0a6-598f10a7283c)


In oppia\core\templates\pages\blog-dashboard-page\blog-card\blog-card.component.html
3: Changing width of blog-card class to 100%
Before Changes:
![image](https://github.com/oppia/oppia/assets/155609546/8d7bb119-f191-4d05-9d04-00a737979ce8)
After Changes:
![image](https://github.com/oppia/oppia/assets/155609546/b6d0c63f-cc19-4c6f-ae1e-e6c452f5d7bf)

4:Removing object-fit :cover from class blog-card-thumbnail-container.
Before Changes:
![image](https://github.com/oppia/oppia/assets/155609546/26cef774-b5e4-4f8d-bef7-e1df2db249f5)
After Changes:
![image](https://github.com/oppia/oppia/assets/155609546/73f88d7b-ddae-4872-bb2f-44199a3f02f2)

Finally :
Before Changes done:

https://github.com/oppia/oppia/assets/155609546/27dba60a-b8f3-4018-9768-3e0c5c941f3c

After All Changes done:-

https://github.com/oppia/oppia/assets/155609546/ac152507-effb-447a-af39-0a6722ee1e9d

Please suggest any further improvement

seanlip commented:
@2008sahil You need to get this working on your local machine. Go to /admin > Roles and give yourself blog admin / blog editor permissions. Then go to /blog-dashboard to create blog posts. (See if you can also learn how to use main.py to figure out what routes and pages exist.)

In terms of your screenshots: I actually think that the "before changes" view in point 1 is fine, we should maximize the space if there is not much space. Past a certain point it is fine for the white card to extend closer to the edges of the screen.

2008sahil commented:
@seanlip ,First of all I wanted to express my sincere gratitude for your guidance and support throughout the review process of the pull request.
Here are the changes that i think solve the current behaviour :- 
Changes in oppia\core\templates\pages\blog-home-page\blog-home-page.component.css
1:Changing the padding-left and padding-right to 3rem at line no. 30 and 31 .
2:Changing the width to 410px at line no. 26.
![image](https://github.com/oppia/oppia/assets/155609546/3d45b2ec-94bf-4333-93f3-75f2c32796a9)
Changes in oppia\core\templates\pages\blog-dashboard-page\blog-card\blog-card.component.html
1:Removing object-fit :cover from class blog-card-thumbnail-container.
Final result:-

https://github.com/oppia/oppia/assets/155609546/7a51ffd1-bb50-4a04-84fb-d5f5f1538f0f

Commit:
Resolved responsive issues in blog page

Commit:
Merge branch 'blogfix' of https://github.com/2008sahil/oppia into blogfix

Commit:
solved responsive issues

2008sahil commented:
I am very sorry I think i messed up the things during updating the code to the latest develop branch Please give me some time i will make the things correct.

2008sahil commented:
@seanlip, I made an error while updating my  pull request  and inadvertently corrupted the branch and would like to seek your permission to close it.I've made the necessary corrections in a new branch and plan to submit a fresh pull request. Your guidance in this matter is highly appreciated.

Thank you for your understanding.

2008sahil commented:
I have mistakenly corrupted this branch so I will closing this pr and will create a new pr very soon.

2008sahil commented:
@seanlip ,PTAL

seanlip commented:
@2008sahil Two quick notes based on your video:

1. Why is the page number so displaced from the arrows surrounding it?
2. In the first blog post, the "Oppia" logo hugs the border too tightly, can this be fixed? (Is it due to the image or a lack of padding?)

Commit:
added margin to blog-navigate icon

Commit:
fixed all changes

Commit:
linter issue resolved

2008sahil commented:
@seanlip , I have mistakenly given the margin to the class which was displacing the arrows ,but now i corrected it .
Secondly  "Oppia" logo hugs are border too tightly because the image cropped that way there is no error for that in the code i have changed the image and corrected all the suggested changes Please take a look.

https://github.com/oppia/oppia/assets/155609546/0f9f1e0c-f35c-4886-9b45-12b002a8c6a1


https://github.com/oppia/oppia/assets/155609546/aed5b452-672a-4a3a-9d1b-82724e0b6c29

Commit:
linter issue resolved

Commit:
all linter checks resolved

seanlip commented on core/templates/pages/blog-home-page/blog-home-page.component.html:
```diff
@@ -88,7 +88,7 @@
           <img [src]="getStaticImageUrl('/general/no_explorations_found.png')">
         </picture>
-        <p class="text-center">
+        <p class="text-center no_result_found">
```
Please follow the existing naming convention for class names (we hyphenate them).

seanlip reviewed (changes requested):
@2008sahil Looks pretty good! Only one note from my end.

Commit:
changed naming to no-result-found

2008sahil committed:
Merge branch 'develop' into blogfix

2008sahil commented:
@seanlip , i changed the class name to no-result-found , I hope everything is fine now thankyou for your guidance.

seanlip reviewed (approved):
Thanks, LGTM from my end!

seanlip commented:
FYI you'll need @kevintab95's review too -- but he's travelling, should be back tomorrow. In the meantime feel free to attempt a different issue if you like :) Thanks!

kevintab95 reviewed (approved):
LGTM, thanks @2008sahil!
